GildedRose/Strategy/Stratege.cs

```diff
@@ -0,0 +1,74 @@
+﻿using GildedRose.Stratege;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GildedRose.Strategy
+{
+    public class AgedBrieUpdateStrategy : IUpdateStrategy
+    {
+        public void Update(Item item)
+        {
+            if (item.Quality < 50)
+            {
+                item.Quality++;
+            }
+            item.SellIn--;
+            if (item.SellIn < 0 && item.Quality < 50)
+            {
+                item.Quality++;
+            }
+        }
+    }
+
+    public class BackstagePassesUpdateStrategy : IUpdateStrategy
+    {
+        public void Update(Item item)
+        {
+            if (item.Quality < 50)
+            {
+                item.Quality++;
+                if (item.SellIn < 11 && item.Quality < 50)
+                {
+                    item.Quality++;
+                }
+                if (item.SellIn < 6 && item.Quality < 50)
+                {
+                    item.Quality++;
+                }
```

Comment:
Increase item quality can be an inner method of item and can avoid duplicated code. If the upper limit(50) needs to be changed, you will just need to modify item class